#  Aircraft Risk Analysis for Business Expansion  

## ✈️ Introduction  
As part of the company's diversification strategy, they are exploring the aviation industry. However, operating aircrafts come with inherent risks. To make informed investment decisions, they must assess which aircraft types pose the **lowest operational risk** based on historical accident data.  

In this project, I will analyze **aviation accident data (1962–2023)** from the National Transportation Safety Board (NTSB) to:  
- **Identify risk factors** associated with aircraft accidents.  
- **Determine the safest aircraft types** for commercial and private operations.  
- **Provide actionable insights** to the aviation division for selecting aircraft with minimal risk.  

Using **data cleaning, imputation, analysis, and visualization**, we aim to transform raw aviation data into meaningful business insights.  


## 🔍 Methodology: Approach to Aircraft Risk Analysis  

To ensure a **thorough and accurate** assessment of aircraft risk, this notebook follows a structured approach:  

### 1️⃣ Data Exploration  
- Load and inspect the dataset to understand its structure and key attributes.  
- Identify missing values, inconsistencies, and potential data quality issues.  

### 2️⃣ Data Cleaning & Preprocessing  
- Handle missing values through imputation or removal where necessary.  
- Standardize formats and correct inconsistencies.  
- Filter relevant data to focus on meaningful insights.  

### 3️⃣ Data Visualization  
- Use **charts, graphs, and interactive visualizations** to identify trends and patterns in accident occurrences.  
- Compare accident rates across different aircraft types, years, and other factors.  

### 4️⃣ Risk Analysis & Insights Generation  
- Perform Basic **statistical analysis** to determine the most and least risky aircraft.  
- Identify key **risk factors** influencing accident likelihood.  
- Correlate accident severity with aircraft type, age, and operational conditions.  

### 5️⃣ Business Recommendations  
- Summarize key findings in a clear format.  
- Provide **data-driven recommendations** to the aviation division on the safest aircraft for investment.  

## 【1】*Data Exploration*  

In [16]:
# Import essential libraries for data analysis and visualization  
import pandas as pd                 # Data manipulation and analysis  
import matplotlib.pyplot as plt     # Data visualization  
import seaborn as sns               # Statistical Data visualization  

In [17]:
# Load the dataset and inspect the first few rows
df = pd.read_csv("Data/Aviation_Data.csv")
df.head()

C:\Users\vihaa\AppData\Local\Temp\ipykernel_20580\3669091942.py:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("Data/Aviation_Data.csv")


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

The dataset raised a **DtypeWarning**, meaning some columns contain **mixed data types** (e.g., numbers and text), which can cause issues during analysis; we will address this in the data cleaning stage by inspecting these columns, converting them to appropriate types, and handling inconsistencies.  


As you can see from the output above, this dataset is **very extensive**, containing numerous columns with detailed aviation accident records. However, to effectively answer our **business question**, we must focus only on the most relevant data points. By narrowing down our selection, we ensure that our analysis remains **targeted, efficient, and data-driven** in making a well-informed decision.  

### **Key Columns We Will Use:**  

- **`Make` & `Model`** → Identifies aircraft manufacturers and specific models to compare safety records.  
- **`Event.Date`** → Allows us to analyze accident trends over time.  
- **`Purpose.of.flight`** → Filters for **business flights**, since our company is focused on commercial aviation.  
- **`Broad.phase.of.flight`** → Helps identify when accidents are most likely to occur (e.g., takeoff, cruise, landing).  
- **`Total.Fatal.Injuries`, `Total.Serious.Injuries`, `Total.Minor.Injuries`, `Total.Uninjured`** → Quantifies accident severity and survivability.  

### **Why These Columns?**  
These fields provide **direct insights into aircraft risk levels** by showing:  
✅ Which **aircraft types** are more prone to accidents.  
✅ The **phases of flight** where accidents occur most often.  
✅ How accident rates have **changed over time**.  
✅ Whether people **walk away unharmed** or if accidents are severe.  

### **Note:**  
Since this is a **U.S.-based company**, we will focus only on accidents that occurred **within the United States**. As we progress through this notebook, we will filter our dataset accordingly to remove irrelevant data and improve analysis accuracy.  


In [18]:
# Check the dimensions of the dataset (rows, columns)
df.shape

(90348, 31)

In [19]:
# View summary statistics
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [20]:
# View a summary of the dataset columns
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [21]:
# Check for missing values in each column
df.isnull().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40216
Airport.Name              37644
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2841
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8555
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7843
Publication.Date          16689
dtype: i

From the output above, we can see that many columns contain a significant number of missing values. This presents a challenge because incomplete data can lead to inaccurate visualizations and misleading analysis.  

### Data Exploration Summary  

Our dataset is extensive but contains many missing values and inconsistencies, requiring careful cleaning before analysis. We've identified key columns relevant to our business question, including aircraft make and model, injury severity, phase of flight, and accident causes. Additionally, we will filter the data to focus on U.S. flights as our company operates domestically.  

In the next section, we will clean the data for accurate visualization and analysis.  


## 【2】*Data Cleaning & Preprocessing*  


###  Why Cleaning Is Necessary?  
- **Ensures Data Accuracy** → Missing values can distort trends and patterns.  
- **Improves Visualization** → Charts and graphs require complete data for meaningful insights.   

To ensure our final insights are reliable and data-driven, we need to properly handle these missing values.  

Here we will:  
✅ Drop columns with excessive missing values.  
✅ Fill or impute missing values where necessary.  
✅ Filter and refine the dataset for meaningful analysis.  

By carefully cleaning the data, we set the foundation for accurate **visualization and decision-making**.


### Filtering For U.S. Accidents 

In [22]:
# Filter the dataset to include only accidents that occurred in the United States
df_clean = df[df["Country"] == "United States"].copy()

print(f'Number of records before filtering:{df.shape}\nNumber of records after filtering:{df_clean.shape}')

Number of records before filtering:(90348, 31)
Number of records after filtering:(82248, 31)


We filtered the dataset to include only accidents that occurred in the United States. This reduced the number of records from **90,348** to **82,248**, confirming that the filtering worked as expected.


### Filtering For Airplanes 

In [23]:
df_clean[df_clean["Aircraft.Category"] == "Airplane"]

Event.Id Investigation.Type Accident.Number  Event.Date  \
5      20170710X52551           Accident      NYC79AA106  1979-09-17   
7      20020909X01562           Accident      SEA82DA022  1982-01-01   
8      20020909X01561           Accident      NYC82DA015  1982-01-01   
12     20020917X02148           Accident      FTW82FRJ07  1982-01-02   
13     20020917X02134           Accident      FTW82FRA14  1982-01-02   
...               ...                ...             ...         ...   
90328  20221213106455           Accident      WPR23LA065  2022-12-13   
90332  20221215106463           Accident      ERA23LA090  2022-12-14   
90335  20221219106475           Accident      WPR23LA069  2022-12-15   
90336  20221219106470           Accident      ERA23LA091  2022-12-16   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   

               Location        Country   Latitude  Longitude Airport.Code  \
5            BOSTON, MA  United States  42.445277 -70.758333          NaN   
7           PULLMAN, WA  United States        NaN        NaN          NaN   
8      EAST HANOVER, NJ  United States        NaN        NaN          N58   
12            HOMER, LA  United States        NaN        NaN          NaN   
13           HEARNE, TX  United States        NaN        NaN          T72   
...                 ...            ...        ...        ...          ...   
90328     Lewistown, MT  United States    047257N   0109280W         KLWT   
90332      San Juan, PR  United States    182724N   0066554W          SIG   
90335       Wichita, KS  United States    373829N   0972635W          ICT   
90336   Brooksville, FL  United States    282825N   0822719W          BKV   
90345        Payson, AZ  United States    341525N   1112021W          PAN   

                         Airport.Name  ... Purpose.of.flight  \
5                                 NaN  ...               NaN   
7                  BLACKBURN AG STRIP  ...          Personal   
8                             HANOVER  ...          Business   
12                                NaN  ...          Personal   
13                   HEARNE MUNICIPAL  ...          Personal   
...                               ...  ...               ...   
90328      Lewiston Municipal Airport  ...               NaN   
90332   FERNANDO LUIS RIBAS DOMINICCI  ...          Personal   
90335  WICHITA DWIGHT D EISENHOWER NT  ...               NaN   
90336      BROOKSVILLE-TAMPA BAY RGNL  ...          Personal   
90345                          PAYSON  ...          Personal   

                         Air.carrier Total.Fatal.Injuries  \
5                         Air Canada                  NaN   
7                                NaN                  0.0   
8                                NaN                  0.0   
12                               NaN                  0.0   
13                               NaN                  1.0   
...                              ...                  ...   
90328                            NaN                  0.0   
90332  SKY WEST AVIATION INC TRUSTEE                  0.0   
90335                            NaN                  0.0   
90336               GERBER RICHARD E                  0.0   
90345                            NaN                  0.0   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
5                        NaN                  1.0            44.0   
7                        0.0                  0.0             2.0   
8                        0.0                  0.0             2.0   
12                       0.0                  1.0             0.0   
13                       0.0                  0.0             0.0   
...                      ...                  ...             ...   
90328                    0.0                  0.0             1.0   
90332                    0.0                  0.0             1.0   
90335                    0.0                  0.0             1.0   
90336                    1.0   

Filtering by "Aircraft.Category" reduces records from **82,248 to 24,229**, indicating missing or misclassified data. We need a better approach to accurately filter airplane-related records.

In [24]:
df_clean["Aircraft.Category"].value_counts(dropna=False)

Aircraft.Category
NaN                  54094
Airplane             24229
Helicopter            2723
Glider                 503
Balloon                229
Gyrocraft              172
Weight-Shift           161
Powered Parachute       90
Ultralight              25
WSFT                     9
Unknown                  4
Blimp                    4
Powered-Lift             3
Rocket                   1
ULTR                     1
Name: count, dtype: int64

In [25]:
df["Country"].value_counts(dropna=False)

Country
United States               82248
NaN                          1685
Brazil                        374
Canada                        359
Mexico                        358
                            ...  
Mauritania                      1
Pacific Ocean                   1
Obyan                           1
Guernsey                        1
Turks and Caicos Islands        1
Name: count, Length: 220, dtype: int64

Why did our Airplane filter fail?  

- Filtering for U.S. accidents worked well because **only 1,685 out of 90,348** entries in the "Country" column were missing.  
- In contrast, **54,094 out of 82,248** entries in "Aircraft.Category" are missing—over **65% of the data**.  

Since filtering this way would drop most records, we take a different approach:  
For rows where "Aircraft.Category" is missing, we compare the "Make" column to an **FAA airplane registry** to filter accurately.

In [26]:
FAA_df = pd.read_csv("Data/FAA_Airplane_Registry.csv")

# Extract unique manufacturer names, drop NaN values, and convert to lowercase
airplane_manufacturers = FAA_df["mfr_name"].dropna().str.lower().unique().tolist()
airplane_manufacturers

['999 leasing ii llc',
 'a r b c inc dba lindstrand',
 'ab sportine aviacija',
 'aero commander',
 'aero vodochody',
 'aerofab inc',
 'aerojones aviation usa llc',
 'aeronca',
 'aeropilot s r o',
 'aeroprakt manufacturing sp zoo',
 'aeropro cz s r o',
 'aeros',
 'aerospatiale',
 'aerospatiale/alenia',
 'aerostar international inc',
 'aerotek',
 'aeroworks',
 'agusta spa',
 'agustawestland philadelphia',
 'agustawestland philadelphia co',
 'air fly balonismo e inflaveis',
 'air global jets llc',
 'air res aviation sp z o o',
 'air tractor inc',
 'airborne windsports pty ltd',
 'airbus',
 'airbus canada lp',
 'airbus helicopters',
 'airbus helicopters deutschland',
 'airbus helicopters inc',
 'airbus industrie',
 'airbus s a s',
 'airbus sas',
 'aircraft mfg & development co',
 'albano jr thomas j',
 'alex bantum',
 'alexander schleicher gmbh & co',
 'alisport srl',
 'alisport swiss s a g l',
 'alon',
 'am aviation llc',
 'american',
 'american aviation',
 'american champion aircraft',
 

In [27]:
# Convert the Make column to lowercase for accurate comparability
df_clean["Make"] = df_clean["Make"].str.lower()
df_clean.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [28]:
# Filter for known airplanes based on Aircraft.Category OR Make column
df_clean = df_clean[
    (df_clean["Aircraft.Category"] == "Airplane") |
    (df_clean["Aircraft.Category"].isna() & df_clean["Make"].isin(airplane_manufacturers))
].copy()
df_clean

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
5      20170710X52551           Accident      NYC79AA106  1979-09-17   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
5           BOSTON, MA  United States  42.445277 -70.758333          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
5              NaN  ...               NaN          Air Canada   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
4                      1.0                    2.0                  NaN   
5                      NaN                    NaN                  1.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
4                 0.0               VMC               Approach   
5                44.0               VMC                  Climb   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

By using the FAA registry to identify airplane manufacturers instead of relying solely on the "Aircraft.Category" column, we increase our valid entries from **24,229 to 68,266**.  

This ensures we retain significantly more relevant data while still focusing only on airplanes.  


### Filtering For Business Flights 

In [29]:
df_clean['Purpose.of.flight'].value_counts(dropna=False)

Purpose.of.flight
Personal                     40827
Instructional                 9161
Unknown                       4665
Business                      3387
Aerial Application            3238
NaN                           2007
Positioning                   1167
Other Work Use                 850
Ferry                          600
Public Aircraft                588
Aerial Observation             520
Executive/corporate            442
Flight Test                    266
Skydiving                      166
Banner Tow                     100
Public Aircraft - Federal       62
Air Race show                   61
Glider Tow                      40
Public Aircraft - State         32
Air Race/show                   28
Public Aircraft - Local         19
Firefighting                    16
External Load                   10
ASHO                             6
Air Drop                         5
PUBS                             3
Name: count, dtype: int64

In [30]:
df_clean = df[df["Purpose.of.flight"] == "Business"].copy()
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4018 entries, 8 to 90326
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                4018 non-null   object 
 1   Investigation.Type      4018 non-null   object 
 2   Accident.Number         4018 non-null   object 
 3   Event.Date              4018 non-null   object 
 4   Location                4017 non-null   object 
 5   Country                 4005 non-null   object 
 6   Latitude                929 non-null    object 
 7   Longitude               929 non-null    object 
 8   Airport.Code            2099 non-null   object 
 9   Airport.Name            2261 non-null   object 
 10  Injury.Severity         4014 non-null   object 
 11  Aircraft.damage         3926 non-null   object 
 12  Aircraft.Category       947 non-null    object 
 13  Registration.Number     3965 non-null   object 
 14  Make                    4018 non-null   obje

### Drop Irrelevant Columns

In [31]:
# Selecting only the relevant columns
columns_to_keep = [
    "Make", "Model", "Event.Date", "Purpose.of.flight", "Broad.phase.of.flight",
    "Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries",
    "Total.Uninjured", "Aircraft.Category", "Country"
]

# Updating df_clean to keep only the relevant columns
df_clean = df_clean[columns_to_keep]

# Verify the updated dataframe structure
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4018 entries, 8 to 90326
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Make                    4018 non-null   object 
 1   Model                   4016 non-null   object 
 2   Event.Date              4018 non-null   object 
 3   Purpose.of.flight       4018 non-null   object 
 4   Broad.phase.of.flight   3373 non-null   object 
 5   Total.Fatal.Injuries    3673 non-null   float64
 6   Total.Serious.Injuries  3589 non-null   float64
 7   Total.Minor.Injuries    3608 non-null   float64
 8   Total.Uninjured         3825 non-null   float64
 9   Aircraft.Category       947 non-null    object 
 10  Country                 4005 non-null   object 
dtypes: float64(4), object(7)
memory usage: 376.7+ KB


In [32]:
# Dropping additional columns that have already served their purpose
df_clean = df_clean.drop(columns=["Country", "Aircraft.Category"])
df_clean.info()


<class 'pandas.core.frame.DataFrame'>
Index: 4018 entries, 8 to 90326
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Make                    4018 non-null   object 
 1   Model                   4016 non-null   object 
 2   Event.Date              4018 non-null   object 
 3   Purpose.of.flight       4018 non-null   object 
 4   Broad.phase.of.flight   3373 non-null   object 
 5   Total.Fatal.Injuries    3673 non-null   float64
 6   Total.Serious.Injuries  3589 non-null   float64
 7   Total.Minor.Injuries    3608 non-null   float64
 8   Total.Uninjured         3825 non-null   float64
dtypes: float64(4), object(5)
memory usage: 313.9+ KB


### Handling Missing Values

*Handling Missing Values in the Injuries & Uninjured Columns*

In [33]:
df_clean.describe()

Total.Fatal.Injuries  Total.Serious.Injuries  Total.Minor.Injuries  \
count           3673.000000             3589.000000           3608.000000   
mean               0.629730                0.245472              0.306541   
std                1.290829                0.717425              0.865980   
min                0.000000                0.000000              0.000000   
25%                0.000000                0.000000              0.000000   
50%                0.000000                0.000000              0.000000   
75%                1.000000                0.000000              0.000000   
max               16.000000               13.000000             17.000000   

       Total.Uninjured  
count      3825.000000  
mean          1.691765  
std           6.587232  
min           0.000000  
25%           0.000000  
50%           1.000000  
75%           2.000000  
max         274.000000

*Why are most the quartiles 0?*

The quartiles (25%, 50%, and 75%) are all showing 0.0 probably because a significant number of values in these columns are likely 0. Let's investigate further:

In [34]:
# Check Value Counts
df_clean[["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured"]].apply(pd.Series.value_counts)


Total.Fatal.Injuries  Total.Serious.Injuries  Total.Minor.Injuries  \
0.0                  2609.0                  3034.0                2964.0   
1.0                   438.0                   349.0                 388.0   
2.0                   316.0                   141.0                 155.0   
3.0                   165.0                    41.0                  52.0   
4.0                    76.0                    13.0                  29.0   
5.0                    31.0                     5.0                  10.0   
6.0                    19.0                     1.0                   4.0   
7.0                     3.0                     2.0                   1.0   
8.0                     8.0                     1.0                   2.0   
9.0                     1.0                     1.0                   NaN   
10.0                    4.0                     NaN                   1.0   
11.0                    1.0                     NaN                   NaN   
12.0                    NaN                     NaN                   1.0   
13.0                    NaN                     1.0                   NaN   
14.0                    1.0                     NaN                   NaN   
15.0                    NaN                     NaN                   NaN   
16.0                    1.0                     NaN                   NaN   
17.0                    NaN                     NaN                   1.0   
19.0                    NaN                     NaN                   NaN   
22.0                    NaN                     NaN                   NaN   
38.0                    NaN                     NaN                   NaN   
44.0                    NaN                     NaN                   NaN   
81.0                    NaN                     NaN                   NaN   
115.0                   NaN                     NaN                   NaN   
123.0                   NaN                     NaN                   NaN   
127.0                   NaN                     NaN                   NaN   
158.0                   NaN                     NaN                   NaN   
274.0                   NaN                     NaN                   NaN   

       Total.Uninjured  
0.0               1479  
1.0                980  
2.0                674  
3.0                268  
4.0                174  
5.0                 95  
6.0                 48  
7.0                 31  
8.0                 22  
9.0                 17  
10.0                10  
11.0                 5  
12.0                 2  
13.0                 2  
14.0                 1  
15.0                 2  
16.0                 1  
17.0                 4  
19.0                 1  
22.0                 1  
38.0                 1  
44.0                 1  
81.0                 1  
115.0                1  
123.0                1  
127.0                1  
158.0                1  
274.0                1

From this, we observe that the majority of entries in each injury-related column are 0, making it the most common value. Additionally, there are cases where all four columns have values, some have partial values, or only one column is filled. However, rows where all four columns are missing provide no useful information and should be dropped. For all other missing entries, we will replace them with 0.0 to maintain consistency in the dataset.

In [39]:
# Identifying rows where all four injury-related columns have missing values
df_clean[df_clean[["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured"]].isnull().all(axis=1)]
df_clean


Make       Model  Event.Date Purpose.of.flight  \
8                  Cessna        401B  1982-01-01          Business   
25                 Cessna        414A  1982-01-03          Business   
36                 Cessna         206  1982-01-04          Business   
37                 Navion           A  1982-01-05          Business   
40               Rockwell         114  1982-01-05          Business   
...                   ...         ...         ...               ...   
90224               PIPER  PA46-500TP  2022-11-09          Business   
90245           FAIRCHILD    SA227-AT  2022-11-15          Business   
90263              CESSNA         207  2022-11-20          Business   
90293         LEARJET INC          45  2022-11-29          Business   
90326  AIRBUS HELICOPTERS   EC 130 T2  2022-12-12          Business   

      Broad.phase.of.flight  Total.Fatal.Injuries  Total.Serious.Injuries  \
8                   Landing                   0.0                     0.0   
25                 Approach                   8.0                     0.0   
36                     Taxi                   1.0                     0.0   
37                   Cruise                   3.0                     0.0   
40                   Cruise                   0.0                     0.0   
...                     ...                   ...                     ...   
90224                   NaN                   2.0                     0.0   
90245                   NaN                   0.0                     2.0   
90263                   NaN                   0.0                     0.0   
90293                   NaN                   0.0                     2.0   
90326                   NaN                   0.0                     1.0   

       Total.Minor.Injuries  Total.Uninjured  
8                       0.0              2.0  
25                      0.0              0.0  
36                      1.0              0.0  
37                      0.0              0.0  
40                      0.0              1.0  
...                     ...              ...  
90224                   0.0              0.0  
90245                   0.0              1.0  
90263                   0.0              7.0  
90293                   0.0              6.0  
90326                   0.0              0.0  

[4012 rows x 9 columns]

In [36]:
# Drop rows where all four injury-related columns have missing values
df_clean = df_clean[~df_clean[["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured"]].isnull().all(axis=1)]
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4014 entries, 8 to 90326
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Make                    4014 non-null   object 
 1   Model                   4012 non-null   object 
 2   Event.Date              4014 non-null   object 
 3   Purpose.of.flight       4014 non-null   object 
 4   Broad.phase.of.flight   3372 non-null   object 
 5   Total.Fatal.Injuries    3673 non-null   float64
 6   Total.Serious.Injuries  3589 non-null   float64
 7   Total.Minor.Injuries    3608 non-null   float64
 8   Total.Uninjured         3825 non-null   float64
dtypes: float64(4), object(5)
memory usage: 313.6+ KB


In [37]:
# Replace missing values with 0.0 in specific injury-related columns
df_clean.loc[:, ["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured"]] = df_clean[["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured"]].fillna(0.0)
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4014 entries, 8 to 90326
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Make                    4014 non-null   object 
 1   Model                   4012 non-null   object 
 2   Event.Date              4014 non-null   object 
 3   Purpose.of.flight       4014 non-null   object 
 4   Broad.phase.of.flight   3372 non-null   object 
 5   Total.Fatal.Injuries    4014 non-null   float64
 6   Total.Serious.Injuries  4014 non-null   float64
 7   Total.Minor.Injuries    4014 non-null   float64
 8   Total.Uninjured         4014 non-null   float64
dtypes: float64(4), object(5)
memory usage: 313.6+ KB


*Handling Missing Values in the Make and Model Columns*

We’ve encountered an edge case where rows categorized as "airplane" may have missing values in either the `Make` or `Model` column, yet were retained. These rows don’t add value to the analysis, as knowing the category without `Make` or `Model` isn’t useful for answering the business question. Therefore, we decided to drop rows with missing values in these columns to ensure our dataset remains complete and relevant.


In [38]:
# Drop rows where either the 'Make' or 'Model' column has missing values
df_clean = df_clean.dropna(subset=["Make", "Model"])
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4012 entries, 8 to 90326
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Make                    4012 non-null   object 
 1   Model                   4012 non-null   object 
 2   Event.Date              4012 non-null   object 
 3   Purpose.of.flight       4012 non-null   object 
 4   Broad.phase.of.flight   3371 non-null   object 
 5   Total.Fatal.Injuries    4012 non-null   float64
 6   Total.Serious.Injuries  4012 non-null   float64
 7   Total.Minor.Injuries    4012 non-null   float64
 8   Total.Uninjured         4012 non-null   float64
dtypes: float64(4), object(5)
memory usage: 313.4+ KB
